onnxruntime/core/optimizer/transpose_optimizer/api_impl.cc

```diff
@@ -677,23 +674,35 @@ void ApiGraph::MoveOutput(api::NodeRef& src_node, size_t src_idx, api::NodeRef&
 }
 
 void ApiGraph::CopyValueInfo(std::string_view src_name, std::string_view dst_name) {
-  NodeArg* src_arg = graph_.GetNodeArg(std::string(src_name));
-  if (src_arg != nullptr) {
-    NodeArg& dst_arg = graph_.GetOrCreateNodeArg(std::string(dst_name), src_arg->TypeAsProto());
-    const TensorShapeProto* shape = src_arg->Shape();
-    if (shape == nullptr) {
-      dst_arg.ClearShape();
-    } else {
-      dst_arg.SetShape(*shape);
-    }
+  const NodeArg* src_arg = graph_.GetNodeArg(std::string(src_name));
+  if (!src_arg) {
+    return;
+  }
 
-    ORT_THROW_IF_ERROR(dst_arg.UpdateTypeAndShape(*src_arg, /*strict*/ false, /*override_types*/ false, logger_));
+  const TypeProto* src_type = src_arg->TypeAsProto();
+  if (!src_type) {
+    return;
   }
+
+  NodeArg& dst_arg = graph_.GetOrCreateNodeArg(std::string(dst_name), nullptr);
+
+  if (auto* dst_type = dst_arg.TypeAsProto(); dst_type != nullptr) {
+    int32_t src_data_element_type;
```

Comment:
The caller guarantees this so you don't have to do the check if you don't want to.

Reply:
That's good to know. Will leave the check to make debugging easier.